server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -171,7 +171,9 @@ public synchronized void updateSegments(final SegmentInfos infos) throws IOExcep
             // a lower gen from a newly elected primary shard that is behind this shard's last commit gen.
             // In that case we still commit into the next local generation.
             if (incomingGeneration != this.lastReceivedPrimaryGen) {
-                flush(false, true);
+                if (engineConfig.getIndexSettings().isStoreLocalityPartial() == false) {
```

Comment:
this engine will only flush here and on close.  I don't think we have a reason to commit on close as we will sync from store on re-open?  If thats the case lets push this logic to the flush method call and cover both cases?  Please also add a comment to why we are not committing here.